### ISSUE TITLE
[Bug-App]: Scalars synced through `sync_tensorboard=True` cannot be used as alternate x-axes

### ISSUE THREAD

**Issue by ringohoffman** [wandb/wandb#9546]
### Describe the bug

https://github.com/user-attachments/assets/7fbaac81-c527-489e-8eb3-355b54860223

It looks like `wandb` is trying to join the 2 graphs on `Step`, not tensorboard's `global_step`. It seems like one `SummaryWritter` step = one wand `Step`. So `batch_idx` only has even `Step` values (since it is logged first), and `percent_done` only has odd `Step` values. 

Can we specifically support joining on tensorboard's `global_step` if `sync_tensorboard=True` and joining on `Step` doesn't work?

```python

import torch
import torch.utils.tensorboard
import wandb

log_dir = "saved"

wandb.init(
    project="llama3_2_transformer_cloak_1B",
    dir=log_dir,
    job_type="dryrun",
    sync_tensorboard=True,
)

tb_writer = torch.utils.tensorboard.SummaryWriter(log_dir=log_dir)


for batch_idx, percent_done in enumerate(torch.linspace(0, 1, 20)):
    tb_writer.add_scalar("batch_idx", batch_idx, global_step=batch_idx)
    tb_writer.add_scalar("percent_done", percent_done.item(), global_step=batch_idx)

wandb.finish()

```

**Comment by fmamberti-wandb** [comment]
Hi @ringohoffman , thank you for reaching out with your request.

You can use the `define_metric` method to set the `global_step` as the `step_metric` so that is used on the x-axis.

In your case for example you could add:

```
wandb.define_metric("batch_idx", step_metric="global_step")
wandb.define_metric("percent_done", step_metric="global_step")
```

before starting to log your data and the metrics will be plotted against `global_step` by default:
<img width="1244" alt="Image" src="https://github.com/user-attachments/assets/fecfe7e8-b574-4800-bc88-c92ce25df9b7" />

Please let us know if you have any questions on the above.

**Comment by ringohoffman** [comment]
@fmamberti-wandb thanks, I didn't know you could use `define_metrics` like that. But I am dynamically creating my metrics, and all my logged metrics start with a different prefix, and I don't want to have to call this for every single metric I create. Can you also check out this other issue I created to support glob prefixes better support this?

* https://github.com/wandb/wandb/issues/9549

**Comment by fmamberti-wandb** [comment]
Hi @ringohoffman , thanks for sharing this - I've raised a feature request for allowing to support prefix for define metrics. and will keep you posted.

**Comment by adeebislam8** [comment]
> [@fmamberti-wandb](https://github.com/fmamberti-wandb) thanks, I didn't know you could use `define_metrics` like that. But I am dynamically creating my metrics, and all my logged metrics start with a different prefix, and I don't want to have to call this for every single metric I create. Can you also check out this other issue I created to support glob prefixes better support this?
> 
> * [[Feature]: Support prefix glob for `Run.define_metric` #9549](https://github.com/wandb/wandb/issues/9549)

Agreed. It's really inconvenient to call define metric for every single metric
